### ISSUE TITLE
[Documentation Request] - Please go into more detail on TOR usage

### ISSUE THREAD

**Issue by billyjeen22** [sparrowwallet/sparrow#1496]
The front page at www.sparrowwallet.com lists "Built in Tor" as a feature, but then TOR is only sparingly mentioned in the rest of the documentation, and only when configuring a private electrum server with a .onion address. But then searching through github issues, I see that other connections like connections to mempool and coingecko are sent out over TOR.  It would be nice to get a clear statement about when the "Built in TOR" is used, and what "Built in TOR" means. 

I'm switching from Wasabi Wallet, which has its own "built in tor" client, making all of its connections over the TOR network by default, without any additional configuration. 

The questions I'd like to see answered are: 
* Is there an actual built in TOR proxy server/client? Or does "Built in Tor" just mean "Support for a user specified TOR proxy"? 
* Do I need to specify a .onion electrum server for TOR to be used? 
* When using a public server from the built in drop down list, if I do not specify a proxy server, will "Built in TOR" automatically be used for all of my connections? Including to my chosen public electrum server, mempool, coingecko, DNS, or any other server accessed by the sparrow wallet? 
* If I specify a non-TOR generic SOCKS proxy (like for my VPN provider), will the "built in TOR" establish a tunnel over that connection? Or is TOR disabled at that point? Or does the software make the assumption that any proxy server entered IS a tor proxy, and therefore capable of resolving .onion addresses without launching the built in TOR proxy? 
Thanks. 

PS. I understand this isn't a support forum, and while I'd love to get an answer here and now, most importantly I think it would be helpful to other users going forward if TOR usage was more clear in documentation.

**Comment by craigraw** [comment]
This is covered in the documentation as an FAQ: https://sparrowwallet.com/docs/faq.html#how-does-the-proxy-support-work. 

Sparrow does not make the use of Tor mandatory, since the Tor network can be unreliable, and IMO it is unwise to unconditionally tie access to your funds to an unreliable network, however beneficial it is to your privacy.

To answer your questions:
> Is there an actual built in TOR proxy server/client? 

Yes.

> Do I need to specify a .onion electrum server for TOR to be used?

Yes, if you don't configure an external Tor proxy.

> When using a public server from the built in drop down list, if I do not specify a proxy server, will "Built in TOR" automatically be used for all of my connections?

No. If you want this behaviour, configure an external Tor proxy.

> If I specify a non-TOR generic SOCKS proxy (like for my VPN provider), will the "built in TOR" establish a tunnel over that connection? Or is TOR disabled at that point?

The internal Tor is disabled. Most VPN configuration these days is not done via a proxy.

The TLDR of all this is that users who have more stringent security and privacy requirements should run an external Tor proxy and configure it in Sparrow. Ideally, a Tor proxy should be a long lived process for better connectivity, while a Bitcoin wallet should only be run as necessary for better security.

**Comment by craigraw** [comment]
I've added another FAQ here, which should clarify this: https://sparrowwallet.com/docs/faq.html#does-sparrow-require-tor

**Comment by voidastro4** [comment]
I believe the proxy implementation may be deeply flawed.
When ***only*** the tor socks5 proxy is available sparrow fails:

Test Connection:
```
Could not connect:

Failed to resolve address: ****************
```

So it appears DNS doesn't just leak but is never torified.

socks5 has address resolution as part of the protocol...

**Comment by craigraw** [comment]
@voidastro4 Electrum server connections are (required to be) long-lived TCP connections. For the purpose of limiting resource usage on public servers, Sparrow does not open a new connection for every wallet. [As noted in the documentation](https://sparrowwallet.com/docs/best-practices.html#the-intermediate-stage), I recommend running your own server for better privacy. Sparrow offers [several](https://sparrowwallet.com/docs/connect-node.html) [options](https://sparrowwallet.com/docs/server-performance.html) to achieve this.

**Comment by learnwayback** [comment]
I'm also getting "Failed to resolve address" when connecting to fulcrum servers while blocking all non tor socks5 traffic.

**Comment by craigraw** [comment]
@learnwayback What version of Sparrow are you using? As of the latest release, I cannot capture any DNS resolution traffic from Sparrow when it is connecting to an Electrum server using an onion address (as intended).

**Comment by learnwayback** [comment]
@craigraw 2.1.3 latest.
definitive dns capture and "Failed to resolve address" aren't necessarily the same thing, it may sense there is no network and refuse to work or something.

**Comment by craigraw** [comment]
Ah, I think I see the cause. It appears you are not using an onion address, but relying on an exit relay on the Tor network. As per https://sparrowwallet.com/docs/faq.html#how-does-the-proxy-support-work, in this case Sparrow needs to determine whether the Electrum server address resolves to a local network address so it can connect to it directly while still using Tor for all external connections. This is a very common configuration (I use it often myself), and the cause of the DNS lookup you are seeing.

That said, I've improved this in e31aa7fc to
1. Avoid address resolution where a public server has been configured
2. Assume the address is non-local where resolution fails and a proxy has been configured

For private Electrum servers accessed over Tor, I recommend using an onion address which will avoid local address resolution and be more private in general.
